### PR TITLE
FE New User Flow

### PR DESCRIPTION
**Pull request by pcatkins**
- New unit test for partial user create success
- Route protection for /new-user routes
- Get user and create user API calls
- Check for existing user, show dashboard or new-user page accordingly

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/UserInfoEndpoints.cs [hunk 1]
```diff
@@ -34,6 +35,12 @@ public void Register(IEndpointRouteBuilder endpointRouteBuilder)
             await bus.InvokeAsync<string>(new RegisterNewUserCommand(mapper.Map<Managers.Registry.UserProfile>(request.Profile), new Login(login.Value.identityProvider, login.Value.id)));
 
             return TypedResults.Ok();
+        }).WithOpenApi(op =>
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Users/UserInfoEndpoints.cs:38]: @ytqsl, added this for access in `openapi.d.ts`

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/api/user.ts [hunk 2]
```diff
@@ -2,8 +2,28 @@ import { getClient } from "@/api/client";
...
 const getUserInfo = async (): Promise<Components.Schemas.UserProfile | null> => {
-  const client = await getClient();
-  return (await client.GetUserInfo()).data?.userInfo ?? null;
+  try {
+    const client = await getClient();
+    const response = await client.GetUserInfo();
+
+    // Check if the response has data and userInfo property
+    return response?.data?.userInfo ?? null;
+  } catch (error) {
```
Comment (ytqsl) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/api/user.ts:11]: should it be more specific catch on 404?

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/Dashboard.vue [hunk 3]
```diff
@@ -2,11 +2,13 @@
   <v-app>
     <v-navigation-drawer v-model="drawer" floating class="bg-white">
       <v-list>
-        <v-list-item prepend-icon="mdi-account-edit " title="Sandra Adams" subtitle="sandra_a88@gmail.com">
+        <v-list-item prepend-icon="mdi-account-edit " :title="userStore.fullName" subtitle="sandra_a88@gmail.com">
           <template #subtitle>
-            <v-list-item-title><p class="small">test@test.com</p></v-list-item-title>
             <v-list-item-title
```
Comment (ytqsl) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/Dashboard.vue:7]: can we sort out the formatting here? why is this broken again?

**Review by ytqsl — CHANGES_REQUESTED** [review]

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/Dashboard.vue (hunk 3 above)
Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/Dashboard.vue]: Good question. Trying `"htmlWhitespaceSensitivity": "ignore"` and that does not chop up the html tag into multiple lines.

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/api/user.ts (hunk 2 above)
Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/api/user.ts:13]: Yes can do. It doesn't change much as far as handling the error at this point. We still need to identify how we want to handle errors FE moving forward (alert/toast user, log?, etc.). 

Adjusted to add catch `404`

Comment (ytqsl) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/api/user.ts:13]: not exactly - 404 here has a true meaning: that user was not found, hence it is a new user

**Review by ytqsl — APPROVED** [review]

**Review by ytqsl — APPROVED** [review]

**Review by ytqsl — APPROVED** [review]

**Review by ytqsl — APPROVED** [review]

**Review by ytqsl — APPROVED** [review]

**Review by ytqsl — APPROVED** [review]

**Pull request merged**